be/src/runtime/current_thread.h

```diff
@@ -53,20 +53,18 @@ class CurrentThread {
     CurrentThread() { tls_is_thread_status_init = true; }
     ~CurrentThread();
 
-    void commit() {
+    void commit(bool is_ctx_shift) {
         MemTracker* cur_tracker = mem_tracker();
         if (cur_tracker != nullptr) {
             cur_tracker->consume(_cache_size);
-            if (_cache_size > 0) {
-                _allocated_cache_size -= _cache_size;
-            } else {
-                _deallocated_cache_size += _cache_size;
-            }
-            cur_tracker->update_allocation(_allocated_cache_size);
-            cur_tracker->update_deallocation(_deallocated_cache_size);
             _cache_size = 0;
-            _allocated_cache_size = 0;
-            _deallocated_cache_size = 0;
```

Comment:
how to update `_allocated_cache_size ` & `_deallocated_cache_size`  if  `is_ctx_shift ` is false?

Reply:
It's guaranteed by  RAII, the CurrentThreadMemTrackerSetter's destructor will invoke `commit(true)`